Sources/StreamChat/Controllers/DatabaseObserver/BackgroundDatabaseObserver.swift

```diff
@@ -186,22 +190,17 @@ class BackgroundDatabaseObserver<Item, DTO: NSManagedObject> {
     /// This method is intended to be called from the `managedObjectContext` that is publishing the changes (The one linked to the `NSFetchedResultsController`
     /// in this case).
     /// Once the objects are mapped, those are sorted based on `sorting`
-    private func mapItems(completion: @escaping ([Item]) -> Void) {
+    private func mapItems(_ changes: [ListChange<Item>]?, completion: @escaping ([Item]) -> Void) {
         let objects = frc.fetchedObjects ?? []
-
-        var items: [Item?] = []
-        items = objects.map { [weak self] in
-            guard self?.isDeletingDatabase == false else { return nil }
-            return try? self?.itemCreator($0)
-        }
-
-        let sorting = self.sorting
-        queue.async {
-            var result = items.compactMap { $0 }
-            if !sorting.isEmpty {
-                result = result.sort(using: sorting)
-            }
-            completion(result)
-        }
+        let items = DatabaseItemConverter.convert(
```

Comment:
Can't we support this on the regular Database observers as well? 🤔

Reply:
Since we turned on background mapping, then I did not bother update those (we use `BackgroundDatabaseObserver` over `EntityDatabaseObserver`).
Other ones we'll delete at some point.

Reply:
I see, makes sense 👍